gdal/ogr/ogrsf_frmts/s57/s57reader.cpp

```diff
@@ -1506,7 +1506,30 @@ OGRFeature *S57Reader::ReadVector( int nFeatureId, int nRCNM )
                              poRecord->GetIntSubfield("VRPT",iField,
                              "MASK",iSubField) );
     }
-
+	
+/* -------------------------------------------------------------------- */
+/*      Geometric attributes                                            */
+/*      Retrieve POSACC and QUAPOS attributes                           */
+/* -------------------------------------------------------------------- */
+
+	const int posaccField = poRegistrar->FindAttrByAcronym("POSACC");
+	const int quaposField = poRegistrar->FindAttrByAcronym("QUAPOS");
+	
+	DDFField * field = poRecord->GetField(i);
+	for( int j = 0; j < field->GetRepeatCount(); j++ ) {
+		// POSACC field
+		if (poRecord->GetIntSubfield("ATTV",0,"ATTL",j) == posaccField) {
```

Comment:
Perhaps store the result of poRecord->GetIntSubfield("ATTV",0,"ATTL",j) in a variable so it can be reused for both tests